src/transformers/quantizers/quantizer_gptq.py

```diff
@@ -73,9 +90,13 @@ def _process_model_before_weight_loading(self, model: "PreTrainedModel", **kwarg
             raise RuntimeError("We can only quantize pure text model.")
 
         if self.pre_quantized:
-            model = self.optimum_quantizer.convert_model(model)
+            model = self.optimum_quantizer.convert_model(model, **kwargs)
 
     def _process_model_after_weight_loading(self, model: "PreTrainedModel", **kwargs):
+        # Only with auto-gptq do not support CPU, we should move the model to cuda if available.
+        if model.device.type == "cpu" and not is_gptqmodel_available() and torch.cuda.is_available():
+            model = model.to(0)
+            model.hf_device_map = {"": 0}
```

Comment:
Let's do that in `update_device_map` function. We should only need to modify the `device_map` and not move the model by ourselves.